Marc's stuff

nandezer opened the pull request:
I made the class of the WritingStyleAnalyzer, I called WAnalyzerS.java.

Functions made: getSentenceLengthProbabilities and getWordLengthProbabilities.

nandezer committed:
getSentenceLengthProbabilities function

Get the probabilities for sentences of length [0..longest sentence in the text].

nandezer committed:
getSentenceLengthProbabilities function

Add some comment

nandezer committed:
getWordLengthProbabilities function

Get the probabilities for words of length [0..longest word in the text].

And correct some code mistake from getSentenceLengthProbabilities

nandezer committed:
Update WritingStyleAnalyzer.java

Correct code mistakes

nandezer committed:
Create class for WritingStyleAnalyzer

nandezer committed:
Update WritingStyleAnalyzer.java

interface class, deleting the code
(rewriting it in WAnalyzerS.java)

nandezer committed:
Update WAnalyzerS.java

Checking functions getWordLengthProbabilities and getSentenceLengthProbabilities
(adding useless code in the other functions)

nandezer committed:
Update WAnalyzerS.java

Still testing functions getSentenceLengthProbabilities and getWordLengthProbabilities

nandezer committed:
Update WAnalyzerS.java

Improting libraries

nandezer committed:
Update WAnalyzerS.java

Error while creating the lists (needed to be  "new ArrayList<>()")

nandezer committed:
Update WAnalyzerS.java

Fixing bugs

nandezer committed:
Update WAnalyzerS.java

Fixing bugs

nandezer committed:
Update WAnalyzerS.java

missing a ;

nandezer committed:
Update WAnalyzerS.java

Missing  throws IOException in functions

nandezer committed:
Add function getSentencesPerParagraphProbabilities

Code getSentencesPerParagraphProbabilities and getSentenceLengthProbabilities

nandezer committed:
Update WAnalyzerS.java

Putting the code more "clear" (tab spaces, comments....)

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

take off the IOException

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Create Humanizer.java

nandezer committed:
Delete Humanizer.java

nandezer committed:
getMispellingWordsProbabilities

Starting to do the function getMispellingWordsProbabilities. Some problems with how to find misspelled words

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

carlthome commented:
1. We're actually not using GitHub Flow anymore but in this case I appreciate it. :+1: 
2. The class you've written is pretty messy. You have dead code, weird naming conventions and also changed the interface to perform extraneous IO. I do not support this. Please do general improvements both coding style-wise and functionality-wise. If you have a method to calculate a probability density function of misspellings then you also should construct a humanize function that takes the function and a text as input and returns a purposefully misspelt text.

nandezer committed:
getMispellingWordsProbabilities function

Add in the function getMispellingWordsProbabilities the misspelling check (by checking if the word has synonyms).

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
getMispellingWordsProbabilities function

Debugging the //Check for synonyms part

nandezer committed:
Update WAnalyzerS.java

@Override functions from WritingStyleAnalyzer

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WritingStyleAnalyzer.java

nandezer committed:
Update WAnalyzerS.java

Changing header of functions (taking Path text as a private variable)
an debugging getMispellingWordsProbabilities

nandezer committed:
Update WAnalyzerS.java

Fixing the overide bug

nandezer committed:
Update WAnalyzerS.java

Fixing @override bug

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Debbuging getMispellingWordsProbabilities()

nandezer committed:
getMispellingWordsProbabilities function

Debbuging getMispellingWordsProbabilities the "check for synonyms part"

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

adding ;

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

adding ;

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
getMispellingWordsProbabilities function

fixing bug

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

Adding return after the "catch" clause

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

fixing return

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WAnalyzerS.java

WordNetSynonyms WordSynonym

nandezer committed:
Update WordNetSynonyms.java

create constructor method

nandezer committed:
Update WAnalyzerS.java

nandezer committed:
Update WordNetSynonyms.java

Deleting constructor

nandezer committed:
Update WAnalyzerS.java

Bugs fixed

nandezer committed:
Update WAnalyzerS.java

improvements both coding style-wise and functionality-wise

nandezer committed:
getMisspellingWordsProbabilities function

No overriding function getMisspellingWordsProbabilities

nandezer committed:
Update WAnalyzerS.java

overiding function getMisspellingWordsProbabilities

nandezer committed:
Update WritingStyleAnalyzer.java

Writting properly the name of the function: "getMisspellingWordsProbabilities"

nandezer commented:
I did the general improvements both coding style-wise and functionality-wise that you asked

carlthome commented:
Cool. Checking!

nandezer committed:
Create Humanizer.java

carlthome commented:
You're still reading the entire file on every method call. That seems insanely wasteful. You're also mixing naming conventions for variables. In general the code just seems like a very roundabout way of implementing the class. 200 lines, really? Please do general improvements. Sorry.

nandezer committed:
Update WAnalyzerS.java

Optimiating the code (to not read the text in every method) and changing name of variables

nandezer commented:
Tried to optimize the code by reading the text on the first function only (analyze) and getting how many words, sentences and paragraphs has.
Also changed a some variables names.

nandezer committed:
Create words

Creating database for the humanization of the text

nandezer committed:
Update Humanizer.java

Creating the functions: - textHumanizer: humanize the text given a Path. - CheckForPossibleMisspellingWords: search for possible misspelled words.

nandezer committed:
Update Humanizer.java

Fixing bugs

nandezer committed:
Update words

Add explanation

nandezer committed:
Update words

Addind words

nandezer committed:
Update words

adding more words

nandezer committed:
Update words

nandezer committed:
Update words

add more words

nandezer committed:
Update words

add more words

nandezer committed:
Update words

more words

nandezer committed:
Update Humanizer.java

Add function that helps to choose the more suitable misspelled word

nandezer committed:
Update Humanizer.java

Improving the TextHumanizer function

nandezer committed:
Update Humanizer.java

Write the text into a string

nandezer committed:
Update Humanizer.java

Write text with misspelled words into String

nandezer committed:
Update Humanizer.java

Add a TODO

nandezer committed:
Update Humanizer.java

Add misspelling probabilities in the humanizer

nandezer committed:
Update WAnalyzerS.java

Changing the misspelling probability calculation (from sentences to paragraph)

nandezer commented:
Humanizer part mainly finish.
(I want to add an algorithm that depending the probability of misspelling words choose one word or an other; as higher the probability less similarity between words)

Also changed the misspelling probability calculation from words per sentences to words per paragraph since when I humanize the text I read it per paragraphs.

carlthome commented:
@johnPertoft and @nandezer, you both need a function that picks one element from a set based on a probabilitiy distribution. Refactor the code into a generalized method for this purpose. A new class with a static method would work.

nandezer committed:
Update .gitignore

Merge with master

nandezer committed:
Update Humanizer.java

Manual merging

nandezer committed:
Update and rename MarkovChain.java to  MarkovTrainer.java

Hand merging

nandezer committed:
Update Ngram.java

Hand merging

nandezer committed:
Update ReflectionDocumentGenerator.java

Hand Merging

nandezer committed:
Delete Synonyms.java

Hand merging

nandezer committed:
Delete TextSource.java

Hand Merging

nandezer committed:
Update and rename WikipediaArticles.java to Wikipedia.java

nandezer committed:
Update and rename WordNetSynonyms.java to  WordNet.java

Hand merging

nandezer committed:
Create WritingStyle.java

Hand merging

nandezer committed:
Delete WritingStyleAnalyzer.java

Hand merging

nandezer committed:
Update KeywordGenerator.java

Hand merging

nandezer committed:
Update MarkovTextGenerator.java

Hand merging

nandezer committed:
Update SynonymGenerator.java

Hand merging

nandezer committed:
Merge pull request #23 from ai14/master

Pull request from master

nandezer committed:
Update Humanizer.java

After merge from master, rewritting the code

nandezer committed:
Delete  MarkovTrainer.java

nandezer committed:
Delete  WordNet.java

nandezer committed:
Update WAnalyzerS.java

Deleting override and changing some bugs

nandezer committed:
Merge pull request #24 from ai14/master

Pull request from master

nandezer committed:
Update Humanizer.java

Debugging Humanizer

nandezer committed:
Update WAnalyzerS.java

Changing getMisspellingWordsProbabilities() functions
(calculating the probability out of all the words instead of per paragraph)

nandezer committed:
Update WAnalyzerS.java

Correcting return value of function getMisspellingWordsProbabilities

nandezer committed:
Update Humanizer.java

Selection of misspelled words done

nandezer committed:
Update WAnalyzerS.java

Fixing bugs

nandezer committed:
Update App.java

Humanizing the text

nandezer commented:
Humanizing part done (works fine) it's a bit poor the algorithm to select the words though.

nandezer committed:
Update Humanizer.java

Add a todo

nandezer committed:
Update App.java

Problems with the merge, fixing it

carlthome commented on src/main/java/com/github/ai14/prosammgen/Humanizer.java:
```diff
@@ -1,5 +1,159 @@
+     * @param OriginalWord
+     * @return
+     */
+    private double[] getSimilaritudesBetweenWords(List<String> WordsToCompare, String OriginalWord) {
```
What's a similaritude?

carlthome commented on src/main/java/com/github/ai14/prosammgen/Humanizer.java:
```diff
@@ -1,5 +1,159 @@
 public class Humanizer {
-  // TODO https://github.com/ai14/prosammgen/issues/17
+
+    private List<String> Words;
```
Please keep to using camelCase. You're mixing weird conventions. Read: https://en.wikipedia.org/wiki/Naming_convention_%28programming%29#Java and keep to that.

carlthome commented on src/main/java/com/github/ai14/prosammgen/Humanizer.java:
```diff
@@ -1,5 +1,159 @@
+    }
+
+    /**
+     * * This "humanize" the text by making some misspelled words given a text, a misspellingProb (as higher the probability
```
Weird english. Fix.

carlthome commented on src/main/java/com/github/ai14/prosammgen/Humanizer.java:
```diff
@@ -1,5 +1,159 @@
+
+
+    public   List<String> checkForPossibleMisspellingWords(String correctWord){
+        boolean ChapitalLetter = false;
```
What's a chapital?

carlthome commented on src/main/java/com/github/ai14/prosammgen/Humanizer.java:
```diff
@@ -1,5 +1,159 @@
+            String[] WordsParagraph = paragraph[i].split("\\s+");
+            numberOfWords = WordsParagraph.length;
+            //each word
+            for (int j =0; j < numberOfWords; ++j) {
```
Weird formatting.

carlthome commented on src/main/java/com/github/ai14/prosammgen/Humanizer.java:
```diff
@@ -1,5 +1,159 @@
+        System.err.println(numberOfParagraphs);
+        StringBuilder HumanizedText = new StringBuilder();
+        //each paragraph
+        for (int i = 0; i < numberOfParagraphs;++i){
```
Why not just skip the comment and do a foreach loop instead?

carlthome commented on src/main/java/com/github/ai14/prosammgen/Humanizer.java:
```diff
@@ -1,5 +1,159 @@
+            }
+            HumanizedText.append("\n\n");
+        }
+        HumanizedText.append("\\end{document}");
```
What's this? This seems incredibly badly thought out. The humanizer should not perform LaTeX formatting at all!

carlthome commented:
Still seems kinda shaky, everything. Needs general improvements. I think you know yourself when the code is getting mature enough for a merge into master. Please rework it up to that point. Also, don't forget to merge master into your branch as soon as David's additions have been merged into master.

carlthome commented:
Since I've stared at this pull request for quite some time now I'm losing sight of the important stuff. @johnPertoft, could you take over when Marc needs a code review next time?

nandezer committed:
Update Humanizer.java

Change issues from the pull request

nandezer committed:
Update App.java

nandezer committed:
Update WAnalyzerS.java

Adding camelCase.

nandezer committed:
Create keyboardclosecharacters

Adding for each keyboard character a list of their closest characters.

nandezer committed:
Merge pull request #26 from ai14/master

Pull request from master

nandezer committed:
Update words

nandezer committed:
Update Humanizer.java

Avoid change words that are not from an asnwer
(Problems with Tex creation!)

nandezer committed:
Update Humanizer.java

Bug with the Tex generation Fix

nandezer committed:
Merge pull request #27 from ai14/master

Pull request from master

nandezer committed:
Update Humanizer.java

Fixing bug when it comes to write "_"

nandezer committed:
Update Humanizer.java

Bug with the "_" character fix

nandezer committed:
Update Humanizer.java

try Levenshtein distance algorithm

nandezer committed:
Update Humanizer.java

Reducing text, adding and correcting comments

nandezer committed:
Update Humanizer.java

removing prints

nandezer committed:
Merge pull request #31 from ai14/master

Pull request from master

nandezer committed:
Update Humanizer.java

Recommenting

nandezer committed:
Merge pull request #32 from ai14/master

Pull request from master

nandezer committed:
Update Humanizer.java

Adding functionality to skip all words that have Latex Characters (begin{document},....)
Not working yet

nandezer committed:
Update WAnalyzerS.java

Changing the resource source

nandezer committed:
Create keyboardclosecharacters

Adding the File in the correct place

nandezer committed:
Delete keyboardclosecharacters

nandezer committed:
Create words

Relocation file to the proper location

nandezer committed:
Delete words

File relocated some where else

nandezer committed:
Merge pull request #33 from ai14/master

Pull request from master

nandezer committed:
Merge pull request #34 from ai14/master

Pull request from master

nandezer committed:
Update Humanizer.java

Fixing when misspelling a word
Don't misspell a Latex command
